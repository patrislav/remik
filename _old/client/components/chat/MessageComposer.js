import React, {Component, PropTypes} from 'react'

const ENTER_KEY_CODE = 13

export default class MessageComposer extends Component {
  static propTypes = {
    onChange: PropTypes.func,
    onSubmit: PropTypes.func,
  }

  constructor() {
    super()
    this.state = {
      text: ''
    }
  }

  render() {
    return (
      <div className="messageComposer">
        <textarea
          name="chat-message"
          value={this.state.text}
          onChange={this._onChange}
          onKeyDown={this._onKeyDown}
        />
        <button onClick={this._onSubmit}>Send!</button>
      </div>
    )
  }

  _onChange = (event) => {
    this.setState({
      text: event.target.value
    })

    if (this.props.onChange) {
      this.props.onChange()
    }
  }

  _onKeyDown = (event) => {
    if (event.keyCode === ENTER_KEY_CODE) {
      event.preventDefault()
      this._onSubmit()
    }
  }

  _onSubmit = () => {
    let message = this.state.text.trim()
    if (message) {
      this.setState({ text: '' })
      if (this.props.onSubmit) {
        this.props.onSubmit(message)
      }
    }
  }

}
