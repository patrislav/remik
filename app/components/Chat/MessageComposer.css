.composer {
  width: 100%;
  height: 100%;
  display: flex;
}

.input {
  flex: 1;
}

.submit {
  flex: none;
}
