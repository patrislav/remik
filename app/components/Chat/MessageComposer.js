import React, {PropTypes} from 'react'
import keycode from 'keycode'
import Button from '../Button'
import TextArea from '../TextArea'
import styles from './MessageComposer.css'

class MessageComposer extends React.Component {
  static propTypes = {
    onCompose: PropTypes.func
  }

  static defaultProps = {
    onCompose: () => {}
  }

  state = {
    value: ''
  }

  render() {
    const { value } = this.state
    return (
      <div className={styles.composer}>
        <TextArea
          value={value}
          onChange={::this.handleChange}
          onKeyDown={::this.handleKeyDown}
          className={styles.input}
        />
        <Button onClick={::this.handleCompose} className={styles.submit}>Send</Button>
      </div>
    )
  }

  handleChange(value) {
    this.setState({ value })
  }

  handleCompose() {
    const { onCompose } = this.props
    const { value } = this.state
    this.setState({ value: '' })
    onCompose(value)
  }

  handleKeyDown(event) {
    if (event.keyCode === keycode('Enter')) {
      event.preventDefault()
      this.handleCompose()
    }
  }
}

export default MessageComposer
