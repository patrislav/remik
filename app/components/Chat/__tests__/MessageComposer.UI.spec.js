import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import MessageComposer from '../MessageComposer'

describe('<MessageComposer />', () => {
  it('renders correct UI', () => {
    const wrapper = shallow(<MessageComposer />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
