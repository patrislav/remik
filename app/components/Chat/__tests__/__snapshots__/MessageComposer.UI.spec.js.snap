exports[`<MessageComposer /> renders correct UI 1`] = `
<div
  className="composer">
  <TextArea
    className="input"
    onChange={[Function]}
    onKeyDown={[Function]}
    value="" />
  <Button
    className="submit"
    onClick={[Function]}>
    Send
  </Button>
</div>
`;
